itests/kit/node_unmanaged.go

```diff
@@ -23,4 +34,78 @@ type TestUnmanagedMiner struct {
 		PeerID  peer.ID
 		PrivKey libp2pcrypto.PrivKey
 	}
+
+	SealedCids, UnsealedCids map[abi.SectorNumber]cid.Cid
+	SealTickets              map[abi.SectorNumber]abi.SealRandomness
+}
+
+func (tm *TestUnmanagedMiner) ManualOnboardingGeneratePreCommit(
+	ctx context.Context,
+	cacheDirPath,
+	unsealedSectorPath,
+	sealedSectorPath string,
+	sectorNumber abi.SectorNumber,
+	sealRandEpoch abi.ChainEpoch,
+	proofType abi.RegisteredSealProof,
+) {
+	if tm.withMockProofs {
+		tm.SealedCids[sectorNumber] = cid.MustParse("bagboea4b5abcatlxechwbp7kjpjguna6r6q7ejrhe6mdp3lf34pmswn27pkkiekz")
+		return
+	}
+
+	req := require.New(tm.t)
+	tm.t.Logf("Generating proof type %d PreCommit ...", proofType)
+
+	_ = os.Mkdir(cacheDirPath, 0755)
+	unsealedSize := abi.PaddedPieceSize(sectorSize).Unpadded()
+	req.NoError(os.WriteFile(unsealedSectorPath, make([]byte, unsealedSize), 0644))
+	req.NoError(os.WriteFile(sealedSectorPath, make([]byte, sectorSize), 0644))
+
+	tm.t.Logf("Wrote unsealed sector to %s", unsealedSectorPath)
+	tm.t.Logf("Wrote sealed sector to %s", sealedSectorPath)
+
+	head, err := tm.FullNode.ChainHead(ctx)
+	req.NoError(err)
+
+	minerAddrBytes := new(bytes.Buffer)
+	req.NoError(tm.ActorAddr.MarshalCBOR(minerAddrBytes))
+
+	rand, err := tm.FullNode.StateGetRandomnessFromTickets(ctx, crypto.DomainSeparationTag_SealRandomness, sealRandEpoch, minerAddrBytes.Bytes(), head.Key())
+	req.NoError(err)
+	sealTickets := abi.SealRandomness(rand)
+
+	tm.t.Logf("Running proof type %d SealPreCommitPhase1 for sector %d...", proofType, sectorNumber)
+
+	actorIdNum, err := address.IDFromAddress(tm.ActorAddr)
+	req.NoError(err)
+	actorId := abi.ActorID(actorIdNum)
+
+	pc1, err := ffi.SealPreCommitPhase1(
+		proofType,
+		cacheDirPath,
+		unsealedSectorPath,
+		sealedSectorPath,
+		sectorNumber,
+		actorId,
+		sealTickets,
+		[]abi.PieceInfo{},
+	)
+	req.NoError(err)
+	req.NotNil(pc1)
+
+	tm.t.Logf("Running proof type %d SealPreCommitPhase2 for sector %d...", proofType, sectorNumber)
+
+	sealedCid, unsealedCid, err := ffi.SealPreCommitPhase2(
+		pc1,
+		cacheDirPath,
+		sealedSectorPath,
+	)
+	req.NoError(err)
+
+	tm.t.Logf("Unsealed CID: %s", unsealedCid)
+	tm.t.Logf("Sealed CID: %s", sealedCid)
+
+	tm.SealTickets[sectorNumber] = sealTickets
+	tm.SealedCids[sectorNumber] = sealedCid
+	tm.UnsealedCids[sectorNumber] = unsealedCid
```

Comment:
👌 this is great, so we just ask it to make the proof and hold on to it, then for ni-porep we can reuse it for the prove commit, but for the standard path we submit the message.

one thing I don't know yet is whether we need a new cache dir for each run of this; it may be that the per-miner cache dir is not enough, and it should be per-commit

Reply:
@rvagg Yeah we do need a new cache dir for each run. Have added it to the latest commit.